<launch>
  <!-- TODO: remove; Launch Kinect here -->
  <include file="$(find fydp)/launch/robot_config.launch"/>
	
  <arg name="camera" default="camera" />
  <arg name="rgb_ns" default="rgb" />
  <arg name="depth_ns" default="depth_registered" />
  <arg name="image_topic" default="image_rect_color" />
  <arg name="depth_topic" default="image" />
  <arg name="fixed_frame" default="camera_rgb_optical_frame" />
  <arg name="paramfile_classifier" default="$(find face_detector)/param/classifier.yaml" />

  <node name="face_detect_listener" type="face_detect_listener" pkg="fydp" output="screen" >
	<remap from="/face_detect_listener/face_positions" to="/move_base_simple/goal" />
  </node>

  <node name="$(anon dynparam)" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters $(arg camera)/driver">
	<param name="depth_registration" type="bool" value="true" />
  </node>
  
  <!-- http://wiki.ros.org/face_detector -->
  <include file="$(find face_detector)/launch/face_detector_common.launch" >
  	<arg name="camera" value="$(arg camera)" />
  	<arg name="rgb_ns" value="$(arg rgb_ns)" />
  	<arg name="image_topic" value="$(arg image_topic)" />
  	<arg name="depth_ns" value="$(arg depth_ns)" />
  	<arg name="depth_topic" value="$(arg depth_topic)" />
  	<arg name="fixed_frame" value="$(arg fixed_frame)" />
  	
	<arg name="classifier_name" value="frontalface" />
	<arg name="paramfile_classifier" default="$(arg paramfile_classifier)" />
	<arg name="classifier_reliability" value="0.9"/>
	<arg name="do_continuous" value="true" />
	<arg name="do_display" value="true" />
	<arg name="do_publish_faces_of_unknown_size" value="false" /> 
	<arg name="use_rgbd" value="true" />
	<arg name="approximate_sync" value="true" />
	
	<arg name="nodename" value="face_detector" />
  </include>

</launch>
